scss/functions/_get-color.scss

```diff
@@ -1,10 +1,26 @@
 @use '../variables' as *;
 @use 'sass:map';
+// @use 'sass:list';
 
 @function get-color($type, $level) {
   // add the logic here
   // use if else to check if the type and level are valid
   // if they are valid, return the color
   // if they are not valid warn 'Invalid color type or level' and return null
   // null is a special value in Sass that means nothing, and it is used to represent the absence of a value, you need to return null if the type or level is not valid
-}
+  
+  // Guard clause for invalid type 
+  @if map.has-key($colors, $type) == null {
+    @warn 'invalid color type: #{$type}';
+    @return null;
+    }
+  
+  // Guard clause for invalid level
+  @if map.get(map.get($colors, $type), $level) == null { 
+    @warn 'invalid color level: #{$level} for type #{$type}';
+    @return null;
+    }
```

Comment:
line 13 dan 19 masih kurang tepat ya

Reply:
// Guard clause for invalid type 
  @if map.has-key($colors, $type) == null {
    @warn 'invalid color type: #{$type}';
    @return and;
    }
  @if map.has-key(map.get($colors, $type), $level) { 
    @return null;
    }

  // if both type and level are valid, return the color
  @return map.get(map.get($colors, $type), $level);
}

Reply:
sudah saya perbaiki. di revisi 2